### DISCUSSION TITLE
Small issue with: Macro- If Replay Buffer Saved

### DISCUSSION THREAD

**Discussion by Outersquid258** [WarmUpTill/SceneSwitcher#1102] [Q&A] [answered]
Hey, so I have a macro setup as the following.
![image_2024-05-07_205933477](https://github.com/WarmUpTill/SceneSwitcher/assets/169216717/aba5c11e-40c7-471c-ac63-406b73a4e9ba)
The problem is when I save a replay buffer the macro doesn't work the first time after OBS has been started up. Only every time after. I'd be happy for any help I can get lol. Thank you!

**Comment by Outersquid258** [comment]
Oh also I'm just doing these macros to get a sound to play whenever I take a replay, start/ stop recording. If anyone knows a better way to do that you can also let me know 🙃 (also I've seen SRbeep but I think it's outdated or something idk it doesn't work for me)

**Comment by WarmUpTill** [comment, marked as answer]
Thanks for pointing out the issue!
A build with a fix should be available here in a few minutes:
https://github.com/WarmUpTill/SceneSwitcher/actions/runs/9006130135?pr=1099

Let me know if you face any issues with that build.

>Oh also I'm just doing these macros to get a sound to play whenever I take a replay, start/ stop recording. If anyone knows a better way to do that you can also let me know 🙃 (also I've seen SRbeep but I think it's outdated or something idk it doesn't work for me)

Alternatively, you could also simply play a media file with whatever application you have assigned to that file type outside of OBS using the "Run" action.

![image](https://github.com/WarmUpTill/SceneSwitcher/assets/19472752/dec1dc47-83b7-4802-9469-8182b5567844)

Hope that helps! :)

**Reply by Outersquid258** [reply]
Thanks for making a build so fast lol, I really appreciate it! 
Also I did originally try and use the run command to play the sound clips I made, but I don't want a window to popup, so then I made an AutoHotkey script that plays the sound without a window, but for some reason the run macro can't do scripts, since I also tried with .bat and .exe, which work on their own, but not with the macro, so that's why I just went to using the media sources.
It really is a small problem, but I figured I'd mention it if you do want to look into it :p Thanks again!

**Reply by WarmUpTill** [reply]
Sure thing! :)

The plugin should be able to run .bat script or .exe files.
One problem might be that the script is using relative paths and thus files might not be found relative to OBS.
But I would need more details to investigate this further.

Can you maybe share the .bat script you had tried to use in the past?

**Reply by Outersquid258** [reply]
I kinda worded that poorly, I couldn't find out how to make a .bat file play the sound so I just made a test script to see if they would work in the first place, and it didn't. But I just did again to check myself and it did... so I probably just messed something up originally, sry bout that.
I did get autohotkey.exe to work though by renaming it to the script, which is probably why it doesn't work with scene switcher bc of what you said about needing relative paths, but here is a video since I think it's like bad to send .exe files or something. 

https://github.com/WarmUpTill/SceneSwitcher/assets/169216717/c3c2eed2-9988-4b88-a9cd-051090b5eb2a

But at this point I think that just using media sources is a lot more effective than trying to use a script to essentially loop around and play the same .wav file in the end. Also after I figure out how to download the build you just made for the macro will work on the first try I'll be pretty set lol. Thanks for the support again!

**Reply by Outersquid258** [reply]
Oh, actually I did figure out how to download the build you made, I just had to scroll down :I but it doesn't seem to show advanced scene switcher when I actually install it.

**Comment by WarmUpTill** [comment]
> I kinda worded that poorly, I couldn't find out how to make a .bat file play the sound so I just made a test script to see if they would work in the first place, and it didn't. But I just did again to check myself and it did... so I probably just messed something up originally, sry bout that. I did get autohotkey.exe to work though by renaming it to the script, which is probably why it doesn't work with scene switcher bc of what you said about needing relative paths, but here is a video since I think it's like bad to send .exe files or something.
> video.mp4
> 
> But at this point I think that just using media sources is a lot more effective than trying to use a script to essentially loop around and play the same .wav file in the end. Also after I figure out how to download the build you just made for the macro will work on the first try I'll be pretty set lol. Thanks for the support again!

Just guessing, as I don't know what is inside your script:
Does it work, if you press the "Show advanced settings" button of the "Run" action and set the working directory to the path of the folder where you double clicked in in the video?

But as you have working solution, I don't want to waste your time with trying to figure this out, if you prefer to keep it the way you have set it up at the moment! :)



> Oh, actually I did figure out how to download the build you made, I just had to scroll down :I but it doesn't seem to show advanced scene switcher when I actually install it.

Hm .. that seems very strange.
Can you share an OBS log file?

**Reply by Outersquid258** [reply]
Wow I was actually just trying to get it working again. Here you go! https://obsproject.com/logs/yEnXbj0CPMhRdbFo
![image](https://github.com/WarmUpTill/SceneSwitcher/assets/169216717/09bd23f1-5bc4-4c40-80ec-55e26a8ac797)

**Reply by Outersquid258** [reply]
Oh yeah, when I set the working directory it works really well lol. I am actually going to do it like that now, thanks!

**Reply by WarmUpTill** [reply]
> Wow I was actually just trying to get it working again. Here you go! https://obsproject.com/logs/yEnXbj0CPMhRdbFo

That is very strange indeed.
Loading of the library simply failed - not really sure what to make of this error:

> 23:26:54.295: LoadLibrary failed for '../../obs-plugins/64bit/advanced-scene-switcher.dll': A dynamic link library (DLL) initialization routine failed.
23:26:54.295:  (1114)
23:26:54.295: Module '../../obs-plugins/64bit/advanced-scene-switcher.dll' not loaded

>Oh yeah, when I set the working directory it works really well lol. I am actually going to do it like that now, thanks!

Glad to hear that it is working! :)

**Reply by Outersquid258** [reply]
Well the new build you made last night still doesn't work for me. I just had to install the previous stable build, which for some reason works.

**Reply by Outersquid258** [reply]
When I get home I'll see if fresh reinstalling obs will work.

**Reply by WarmUpTill** [reply]
Just to mention it in case you lost your settings:
The last active settings of the plugin are always backed up in the `plugin-config/advanced-scene-switcher` directory.
On Windows you can usually find it under:
`C:\Users\<Your Username Here>\AppData\Roaming\obs-studio\plugin_config\advanced-scene-switcher`
You can then [import](https://github.com/WarmUpTill/SceneSwitcher/wiki/Saving-and-loading-settings) those settings on the General tab.

I am really sorry about that, I am not sure what is causing this type on issue.

**Reply by Outersquid258** [reply]
Nah I made a ton of backups lol. Also I just reinstalled and it still doesn't work for me :/ same statement in the logs too about the libraries not loading.

**Reply by Outersquid258** [reply]
I just tested a bunch to see if all the files from the new build work when I manually install them, but I just leave the "obs-plugins/64bit/advanced-scene-switcher.dll" as from the previous version, and everything loads up but I just get this message
![image](https://github.com/WarmUpTill/SceneSwitcher/assets/169216717/19250a57-a48e-4ec4-841e-eb0b1d531983)
and when I try to add any macro obs just crashes :p sorry for all the trouble I just really want this to work now lol.

**Reply by Outersquid258** [reply]
Ok sorry for all the replies, but I just found the solution thanks to this [thread](https://obsproject.com/forum/threads/solved-issues-with-plugin-loading.79508/). I just needed to install MSVC. And now everything works smoothly! Thank you again!
